src/newshop.cpp

```diff
@@ -1359,11 +1361,19 @@ void shop_list(char *arg, struct char_data *ch, struct char_data *keeper, vnum_t
       else strcpy(buf2, "-");
 
       if (IS_OBJ_STAT(obj, ITEM_NERPS)) {
-        snprintf(ENDOF(buf), sizeof(buf) - strlen(buf), "^Y(N)^n %-29s^n %-6s%2s   %0.2f%c  %9d\r\n", GET_OBJ_NAME(obj),
+        //Format string: "^Y(N)^n %-58s^n %-6s%2s   %0.2f%c  %9d\r\n"
+        //We apply padding for color codes here.
+        snprintf(paddingnumberstr, sizeof(paddingnumberstr), "%d", 58 + count_color_codes_in_string(GET_OBJ_NAME(obj)));
```

Comment:
58 is an extremely magical number here, and I can't tell what it is or where it came from. If it's derived from the length of some spacer string, recommend breaking that spacer string out into its own component and replacing the 58 here with a `strlen(that_string)`.

Reply:
I am using the existing formatting numbers, which were probably arbitrarily derived at some point to a certain width fitting current-at-the-time strings. I am just appending the space that needs appending so that the tables will print properly for the existing strings and factoring in color codes space so columns align properly.

Reply:
I have added the original format strings as comments both for clarity and for reference as I formulated the modifications. There was also 1 (if I remember right, could be 2) char error when it came down to aligning the header with the item prices for non-colored items as well so that's fixed as well.